apps/prairielearn/src/pages/instructorInstanceAdminPublishing/instructorInstanceAdminPublishing.tsx

```diff
@@ -0,0 +1,244 @@
+import assert from 'assert';
+import * as path from 'path';
+
+import sha256 from 'crypto-js/sha256.js';
+import { Router } from 'express';
+import asyncHandler from 'express-async-handler';
+import fs from 'fs-extra';
+import z from 'zod';
+
+import * as error from '@prairielearn/error';
+import { flash } from '@prairielearn/flash';
+import { loadSqlEquiv, queryRows } from '@prairielearn/postgres';
+
+import { PageLayout } from '../../components/PageLayout.js';
+import { CourseInstanceSyncErrorsAndWarnings } from '../../components/SyncErrorsAndWarnings.js';
+import { b64EncodeUnicode } from '../../lib/base64-util.js';
+import { extractPageContext } from '../../lib/client/page-context.js';
+import { CourseInstanceAccessRuleSchema } from '../../lib/db-types.js';
+import { FileModifyEditor, propertyValueWithDefault } from '../../lib/editors.js';
+import { getPaths } from '../../lib/instructorFiles.js';
+import { formatJsonWithPrettier } from '../../lib/prettier.js';
+import { selectUsersAndEnrollmentsByUidsInCourseInstance } from '../../models/enrollment.js';
+import { type CourseInstanceJsonInput } from '../../schemas/infoCourseInstance.js';
+
+import { InstructorInstanceAdminPublishing } from './instructorInstanceAdminPublishing.html.js';
+
+const router = Router();
+const sql = loadSqlEquiv(import.meta.url);
+
+// Validate a list of UIDs against enrollments in this course instance.
+// Returns the list of UIDs that are NOT enrolled (invalidUids).
+router.get(
+  '/extension/check',
+  asyncHandler(async (req, res) => {
+    const {
+      authz_data: { has_course_instance_permission_edit: hasCourseInstancePermissionEdit },
+    } = extractPageContext(res.locals, {
+      pageType: 'courseInstance',
+      accessType: 'instructor',
+    });
+
+    if (!hasCourseInstancePermissionEdit) {
+      throw new error.HttpStatusError(403, 'Access denied (must be course instance editor)');
+    }
+
+    // Accept comma-separated UIDs in query parameter
+    const uidsString = typeof req.query.uids === 'string' ? req.query.uids : '';
+    const uids: string[] = uidsString
+      .split(',')
+      .map((s) => s.trim())
+      .filter((s) => s.length > 0);
+
+    // Verify each UID is enrolled (matches either users.uid or enrollments.pending_uid)
+    const validRecords = await selectUsersAndEnrollmentsByUidsInCourseInstance({
+      uids,
+      courseInstance: res.locals.course_instance,
+      requestedRole: 'Student Data Viewer',
+      authzData: res.locals.authz_data,
+    });
+    const validUids = new Set(validRecords.map((record) => record.user.uid));
+    const invalidUids = uids.filter((uid) => !validUids.has(uid));
+
+    res.json({ invalidUids });
+  }),
+);
```

Comment:
👀 this is unused here, right? Can you move this to the extensions PR so we can review + test this in context?

Reply:
We discussed this is unneeded.